docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -82,30 +82,158 @@ a :ref:`connector-page-sink-provider`.
+
+Type mapping
+^^^^^^^^^^^^
+
+The built-in SQL data types use different Java types as carrier types.
+
+================================== =============
+SQL type                           Java type
+================================== =============
+``BOOLEAN``                        ``boolean``
+``TINYINT``                        ``long``
+``SMALLINT``                       ``long``
+``INTEGER``                        ``long``
+``BIGINT``                         ``long``
+``REAL``                           ``double``
+``DOUBLE``                         ``double``
+``DECIMAL``                        ``long`` for precision up to 19, inclusive;
+                                   ``Int128`` for precision greater than 19
+``VARCHAR``                        ``Slice``
+``CHAR``                           ``Slice``
+``VARBINARY``                      ``Slice``
+``JSON``                           ``Slice``
+``DATE``                           ``long``
+``TIME(P)``                        ``long``
+``TIME WITH TIME ZONE``            ``long`` for precision up to 9;
+                                   ``LongTimeWithTimeZone`` for precision
+                                   greater than 9
+``TIMESTAMP(P)``                   ``long`` for precision up to 6;
+                                   ``LongTimestamp`` for precision greater
+                                   than 6
+``TIMESTAMP(P) WITH TIME ZONE``    ``long`` for precision up to 3;
+                                   ``LongTimestampWithTimeZone`` for precision
+                                   greater than 3
+``INTERVAL YEAR TO MONTH``         ``long``
+``INTERVAL DAY TO SECOND``         ``long``
+``ARRAY``                          ``Block``
+``MAP``                            ``Block``
+``ROW``                            ``Block``
+``IPADDRESS``                      ``[]byte``
+``UUID``                           ``[]byte``
+``HyperLogLog``                    ``Object``
+``P4HyperLogLog``                  ``Object``
+``SetDigest``                      ``Object``
+``QDigest``                        ``Object``
+``TDigest``                        ``Object``
+================================== =============
+
+The ``RecordCursor.getType(int field)`` method returns the SQL type for a field
+and the value for it is returned by one of the following methods, matching
+the carrier type:
+
+* ``getBoolean(int field)``
+* ``getLong(int field)``
+* ``getDouble(int field)``
+* ``getSlice(int field)``
+* ``getObject(int field)``
+
+
+Date and time values of regular precision can be converted into ``long`` using
+static methods from the ``io.trino.spi.type.DateTimeEncoding`` class,
+like ``pack()`` or ``packDateTimeWithZone()``.
+
+UTF-8 encoded strings can be converted to Slices using
+the ``Slices.utf8Slice()`` static method.
+
+.. note::
+
+  ``Slice`` class is provided by the ``io.airlift:slice`` package.
+
+``Int128`` objects for decimal columns can be created using
```

Comment:
Int128 is not just for decimal columns (even though it's currently only used by that type). It's for anything that needs a 128-bit integer number.